react/week2/interactive-todo-list/src/components/CreateTodo.jsx

```diff
@@ -0,0 +1,83 @@
+import React from "react";
+// import { ReactDOM } from "react-dom";
+import { useState } from "react";
+import TodoList from "./TodoList";
+import swal from "sweetalert";
+import Counter from "./Counter";
+
+export default function CreateTodo() {
+  const [todo, setTodo] = useState({ title: "", done: false });
+  const [todoArr, setTodoArr] = useState([]);
+
+  const handleChange = (event) => {
+    let { value } = event.target;
+    let obj = {};
+    obj["title"] = value;
+    obj["done"] = false;
+    setTodo(obj);
+  };
+
+  const createTodo = (event) => {
+    const { name } = event.target;
+    if (event.key === "Enter" || name === "addTodo") {
+      if (todo.title !== "") {
+        const updatedTodoArray = [todo, ...todoArr];
+        setTodoArr(updatedTodoArray);
+        setTodo({ title: "", done: false });
+      } else {
+        swal("Oops", "Please write todo first", "error");
+      }
+    }
+  };
+
+  const completeTodo = (i) => {
+    const updatedTodoArray = [...todoArr];
+    if (updatedTodoArray[i]["done"] !== true) {
+      updatedTodoArray[i]["done"] = true;
+      setTodoArr(updatedTodoArray);
+      swal("Good Job !", "Todo completed", "success");
+    }
```

Comment:
We will now use the `id` to identify the updated todo:

```suggestion
    const updatedTodoArray = [...todoArr];
    const index = updatedTodoArray.findIndex((todo) => todo.id === i);
    if (updatedTodoArray[index]["done"] !== true) {
      updatedTodoArray[index]["done"] = true;
      setTodoArr(updatedTodoArray);
      swal("Good Job !", "Todo completed", "success");
    }
```

Reply:
yes i got the point, thanks a lot